doc/_static/css/custom.css

```diff
@@ -181,6 +210,167 @@ html[data-theme="dark"] .wy-nav-top a {
+    border: none;
+    font-size: 85%;
+    color: var(--dark-link-color);
+}
+
+
+/*
+ * Facelift example page with custom card layout
+ */
+#examples figure {
+  display: inline-flex;
+  align-items: center;
+  flex-direction: column;
+
+  background-color: #eee;
+
+  padding: 24px;
+  padding-bottom: 0;
+
+  /* Tuned so that all cards with up to 3 rows of text (95%) will be the same height. */
+  min-height: 230px;
+
+  margin-right: 4px;
+
+  width: 190px !important;
+
+  border: #ddd 1px solid;
+  border-radius: 8px;
+}
+
+#examples figure figcaption {
+    padding: 16px 0;
+}
+#examples figure figcaption .headerlink {
+    display: none;
+}
+#examples figure figcaption * {
+    margin: 0;
+    display: inline-block;
+}
+#examples figure img {
+    /* Notice: the uniformity of these "cards" is based on the images all being same height */
+    height: 100px;
+
+    border: #bbb 1px solid;
+    border-radius: 8px;
+}
+
+html[data-theme="dark"] #examples figure {
+  background-color: #2b2b2b;
+  border-color: #333;
+}
+html[data-theme="dark"] #examples figure img {
+    border: #111 2px solid;
+}
+
+
+/*
+ * Syntax highlight color fixes for basic code blocks in dark mode.
+ * <code_colors>
+ */
+html[data-theme="dark"] pre {
+    /* baseline */
+    background-color: #282828 !important;
+    color: #e0e0e0 !important;
+}
+html[data-theme="dark"] pre .linenos {
+    /* line numbers */
+    color: #888 !important;
+    border-right-color: #404040 !important;
+}
+html[data-theme="dark"] pre .nc {
+    /* class declarations */
+    color: #0880FF !important;
+}
+html[data-theme="dark"] pre .kc,
+html[data-theme="dark"] pre .kn,
+html[data-theme="dark"] pre .nb,
+html[data-theme="dark"] pre .bp,
+html[data-theme="dark"] pre .k {
+    /* `None` + `import` + typehints + `self` + keywords */
+    color: var(--color-emphasis) !important;
+}
+html[data-theme="dark"] pre .fm,
+html[data-theme="dark"] pre .vm,
+html[data-theme="dark"] pre .nf {
+    /* function + method declarations + `__name__` */
+    color: #00B7FF !important;
+    font-weight: bold;
+}
+html[data-theme="dark"] pre .nd,
+html[data-theme="dark"] pre .ow,
+html[data-theme="dark"] pre .si,
+html[data-theme="dark"] pre .sa {
+    /* decorators + `in` + f-string tokens */
+    color: #D56CF2 !important;
+}
+html[data-theme="dark"] pre .s1,
+html[data-theme="dark"] pre .s2,
+html[data-theme="dark"] pre .sd {
+    /* strings + docstrings */
+    color: #FF873F !important;
+}
+html[data-theme="dark"] pre .c1 {
+    /* comments */
+    color: #4c8831 !important;
+}
+html[data-theme="dark"] pre .o,
+html[data-theme="dark"] pre .mf,
+html[data-theme="dark"] pre .mi {
+    /* operators + numeral variable values */
+    color: #afafaf !important;
+}
+html[data-theme="dark"] pre .gp {
+    /* The REPL `>>>` token */
+    color: var(--color-emphasis) !important;
+}
+html[data-theme="dark"] pre .hll {
+    /* Line highlights background */
+    background-color: #323331 !important;
+}
```

Comment:
Would it be possible to add a comment with a link to the upstream source / doc explaining what these classes do?

Reply:
Certainly would be possible, but honestly I have no idea which system is actually generating this syntax highlighting in the first place. This whole mess was basically just reverse-engineered with the browser inspection tool, which I reckon will be the easiest way in the future as well, as you can just click-select the specific symbol you're interested in currently.